dbt_subprojects/dex/models/trades/ethereum/platforms/pancakeswap_v2_ethereum_base_trades.sql

```diff
@@ -25,6 +25,44 @@ dexs_macro AS (
     }}
 ),
 
+dexs_pcsx AS (
+    -- PancakeSwapX
+    SELECT
+        'pcsx' AS version,
+        a.evt_block_number AS block_number,
+        a.evt_block_time AS block_time,
+        a.swapper AS taker,
+        a.filler AS maker,
+        receive.amount_raw AS token_bought_amount_raw,
+        send.amount_raw AS token_sold_amount_raw,
+        receive.contract_address AS token_bought_address,
+        send.contract_address AS token_sold_address,     
+        a.contract_address AS project_contract_address,
+        a.evt_tx_hash AS tx_hash,
+        a.evt_index
+    
+    FROM {{ source('pancakeswap_ethereum', 'ExclusiveDutchOrderReactor_evt_Fill') }} a 
+
+    LEFT JOIN (
+        select * from {{ source('tokens', 'transfers') }}
+        where blockchain = 'ethereum'
+        and block_date >= date '2024-09-20' 
+        and tx_hash in (select evt_tx_hash from {{ source('pancakeswap_ethereum', 'ExclusiveDutchOrderReactor_evt_Fill') }})
```

Comment:
Do you think we can make this part a CTE with incremental filters before left join?

Reply:
Do you mean like this? 

with transfer as (
  select * from {{ source('tokens', 'transfers') }}
  where blockchain = 'ethereum'
  and block_date >= date '2024-09-20' 
  and tx_hash in (select evt_tx_hash from {{ source('pancakeswap_ethereum', 'ExclusiveDutchOrderReactor_evt_Fill') }})
  {% if is_incremental() %}
  WHERE {{ incremental_predicate('a.evt_block_time') }}
  {% endif %}
)

Reply:
Exactly.